apps/jacket/jacket.star

```diff
@@ -0,0 +1,252 @@
+    # weather_info.append(add_row("Feel", "{0}{1}".format(feels_like, degree_sign)))
+    weather_info.append(
+        render.Row(
+            expanded = True,
+            main_align = "center",
+            cross_align = "center",
+            children = [
+                render.WrappedText("%s" % mainString, "tb-8", align = "center"),
+            ],
+        ),
+    )
+    if show_description != "false":
+        subString = getSubString(feels_like)
+        weather_info.append(render.Box(width = 64, height = 1, color = config.get("divider_color", "#1167B1")))
+        weather_info.append(
+            render.Row(
+                # expanded = True,
+                main_align = "center",
+                cross_align = "center",
+                children = [
+                    render.Marquee(
+                        width = 64,
+                        child = render.Text("%s" % subString, "CG-pixel-3x5-mono"),
+                    ),
+                ],
+            ),
+        )
+
+    return render.Root(
+        child = render.Column(
+            expanded = True,
+            main_align = "space_around",
+            cross_align = "center",
+            children = weather_info,
+        ),
+    )
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "application_key",
+                name = "Application Key",
+                desc = "Ambient Weather Application Key.",
+                icon = "key",
+            ),
+            schema.Text(
+                id = "api_key",
+                name = "API Key",
+                desc = "Ambient Weather API Key.",
+                icon = "key",
+            ),
+            schema.Toggle(
+                id = "show_description",
+                name = "Show Description",
+                desc = "Show description of weather",
+                icon = "gear",
+                default = True,
+            ),
+            schema.Color(
+                id = "divider_color",
+                name = "Divider Color",
+                desc = "The color of the dividers",
+                icon = "brush",
+                default = "#1167B1",
+            ),
+        ],
+    )
+
+def get_weather_data(config):
+    api_key = config.get("api_key", None)
+    application_key = config.get("application_key", None)
+    cached_data = cache.get("weather_data-{0}".format(api_key))
+    if cached_data != None:
+        print("Using existing weather data")
+        cache_res = json.decode(cached_data)
+        return cache_res
+
+    else:
+        if api_key == None:
+            print("Missing api_key")
+            return SAMPLE_STATION_RESPONSE
+        if application_key == None:
+            print("Missing application_key")
+            return SAMPLE_STATION_RESPONSE
+
+        print("Getting new weather data")
+        res = http.get(
```

Comment:
Would it be appropriate to pass `ttl_seconds` here? The default ttl is pretty short at the moment, and may change without warning down the line.